### DISCUSSION TITLE
ability to optimize storage by deleting individual diffs and also cleanin up hitsounds, storyboards etc.

### DISCUSSION THREAD

**Discussion by Inconsist** [ppy/osu#22601] [Ideas]
Can we please bring back the ability to bulk delete individual diffs but not the entire maps? I have like 12500 diffs that are >3 stars which ill likely never play and id like to free up some storage. In stable this was possible, why not in lazer?

**Comment by mk56-spn** [comment]
I very much doubt beatmaps contain a significant amount of data compared to the mp3, beatmap skin, etc which obviously can't be deleted since they are shared across the beatmapset ( almost always ) but I could be wrong

**Reply by monochrome22** [reply]
To help clarify numbers: 
My songs folder is 188GB
.osu files are 1.51GB (0.8%)
Backgrounds are 13.8GB (7.3%)
Video files are 66.8GB (35.5%)
Song files are 80.2GB (42.7%)
Beatmap skin and storyboard files are 25.7GB (13.7%)

Also, #18404

**Reply by mk56-spn** [reply]
right, assuming thats all correct then yeah this would probably be nice to see happen at some point, not a dev though so you'll have to weight till one of them checks this

**Reply by Inconsist** [reply]
I agree individual diffs are more irrelevant but they do reduce the clutter. But my main point would be hitsounds I wont use and skin elements I dont need which take up a significant amount, at least to me. I specifically went to stable and deleted all my star<3  diffs, ran osu cleaner for hitsounds and skin elements, deleted all lazer maps and reimported all maps back into lazer only to realize all sub 3 diffs are there again. Why would that happen? Does it redownload them, or did Stable also just hide my diffs all along?

**Reply by mk56-spn** [reply]
I would be inclined to think that stable is merely faux deleting them in that case. To my knowledge Lazer only currently updates maps with explicit user confirmation ( ie clicking the update button on the beatmaps "panel" )

**Reply by monochrome22** [reply]
Updated my original comment with more accurate numbers.

Couldn't you use star rating filter to hide lower star maps?
